s4e-web/src/app/views/map-view/map/map.component.ts

```diff
@@ -118,6 +116,41 @@ export class MapComponent implements OnInit, OnDestroy {
     });
   };
 
+  public getMapData(): Observable<MapData> {
+    this.working.emit(true);
+    const canvas: HTMLCanvasElement = this.map.getViewport().firstChild as HTMLCanvasElement;
+
+    const r = new ReplaySubject<MapData>(1);
```

Comment:
One time `r` is `request`, another `r` is `mapData$` in case of consistency and readability could you change name of this variable?